server/config.go

```diff
@@ -496,6 +510,12 @@ func CLIFlags() []cli.Flag {
 			Value:   cli.NewStringSlice(),
 			EnvVars: prefixEnvVars("CACHE_TARGETS"),
 		},
+		&cli.BoolFlag{
+			Name:    ReadOnlyFlagName,
+			Usage:   "Whether the proxy should operate in read-only mode.",
+			Value:   false,
+			EnvVars: prefixEnvVars("READ_ONLY"),
+		},
```

Comment:
Why are we adding this? Who is this feature useful for? Can we add some description here or somewhere else as to the intent of this feature.

Reply:
This is a feature included in the v1.5.0 [document](https://docs.google.com/document/d/1vcL2zpmjNijFGyyMTOz3xgjy9zMya9kGcGtNfDK5VS4/edit#task=SZUBJBN_rUxUnced). I think this was requested by celo. The use case is they only need retrieval when running proxy, and it made sense to develop a read mode that does not require a key.

I've modified the usage string to say `Whether to run the proxy in read-only mode; set to true to only allow retrieval from EigenDA.` Please let me know if this is sufficient

Reply:
I think the problem now is that if some op-nodes just want to run the derivation pipeline, there is no way to start the proxy without putting in some fake privateKey.